Add specific updates to extinfo field.

bertptrs opened the pull request:
As the title says, this update adds information about which updates are pending to the extinfo field. This makes warnings much more informational, since it tells you whether you really need to update now or if you can wait for a while.

sumpfralle commented:
Thank you for polishing this plugin up!

You are using a temporary file for storing the data source.
In this case you should use a `trap` for making sure, that the temporary file is removed under all (internal or external) circumstances.
But in this specific case you could simply store the data in a variable and `echo` it into `wc` and `paste`. This would free you from taking care for temporary files.

Additionally: please take care, that you do not introduce additional shell style issues (see `shellcheck`). In your case the `if` expression and the parameters for your `echo` calls lacked quotes.

What do you think?

bertptrs commented:
Great suggestion on using `shellcheck`. I added it to my arsenal.

My main reason for piping to a temp file rather than storing it in memory, is to work around output capture skipping trailing newlines. If I do `updates="$(checkupdates)"`, I cannot distinguish between 0 and 1 updates, since `wc -l` will report 0 lines either way. Piping to file removed this issue. Do you have a suggestion to work around this?

sumpfralle commented:
> My main reason for piping to a temp file rather than storing it in memory, is to work around output capture skipping trailing newlines. If I do updates="$(checkupdates)", I cannot distinguish between 0 and 1 updates, since wc -l will report 0 lines either way.

What an interesting detail! I never thought about that ...

How about checking for an non-empty string `[ -n "$updates" ]` before asking `wc`?

bertptrs commented:
Great suggestion. I've implemented that and your other suggestions. I also removed the explicit warning option since it was dupllicating a munin option.

sumpfralle commented:
Cool - thank you!
